arch/arm/core/swap.S

```diff
@@ -217,6 +220,17 @@ _thread_irq_disabled:
     bx lr
 
 #if defined(CONFIG_ARMV6_M)
+#ifdef CONFIG_IRQ_OFFLOAD
+SECTION_FUNC(TEXT, __svc)
+    push {lr}
```

Comment:
Shouldn't we parse the service call number first? Otherwise this will break if we hit a kernel_oops